src/software/ai/navigator/path_planner/path_planner.h

```diff
@@ -12,11 +14,28 @@
  * if a path exists, otherwise it will return nothing.
  */
 
+using PathType = std::variant<std::vector<Curve>, std::vector<Point>>;
+
 class PathPlanner
 {
    public:
-    virtual std::optional<std::vector<Point>> findPath(const Point &start,
-                                                       const Point &dest) = 0;
+    /**
+     * Returns a path between start and dest.
+     * (consider the case where we have a "StraightLinePathPlanner" for testing purposes)
+     *
+     * @param start start point
+     * @param dest destination point
+     * @param field field
+     * @param obstacles obstacles to avoid
+     *
+     * @return a path between start and dest
+     * 		if no valid path then return an empty vector
+     * 		if spline navigator then return a std::vector of Curve
+     * 		if point navigator then return a std::vector of points
+     * 		    * this vector must include the start point and end point
```

Comment:
I'd shorten this to `@return a path between start and dest`. We don't need to specify what a path is (don't want to duplicate information inherent in the type), and we definitely don't need to specify what "navigator" (I assume you mean "path planner"?) will return what type.

Reply:
I think it's valuable to be specific in this interface, but I agree I shouldn't specific the type of path planner